flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/TableEnvironmentTest.java

```diff
@@ -158,6 +167,121 @@ void testTableFromDescriptor() {
+                .hasMessage(
+                        "Could not execute CreateModel in path `default_catalog`.`default_database`.`M`");
+    }
+
+    @Test
+    void testDropModel() throws Exception {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
+
+        tEnv.createModel("M", TEST_MODEL_DESCRIPTOR);
+
+        final String catalog = tEnv.getCurrentCatalog();
+        final String database = tEnv.getCurrentDatabase();
+        final ObjectPath objectPath = new ObjectPath(database, "M");
+        CatalogModel catalogModel =
+                tEnv.getCatalog(catalog).orElseThrow(AssertionError::new).getModel(objectPath);
+        assertThat(catalogModel).isInstanceOf(CatalogModel.class);
+        assertThat(tEnv.dropModel("M", true)).isTrue();
+        assertThatThrownBy(
+                        () ->
+                                tEnv.getCatalog(catalog)
+                                        .orElseThrow(AssertionError::new)
+                                        .getModel(objectPath))
+                .isInstanceOf(ModelNotExistException.class)
+                .hasMessage("Model '`default_catalog`.`default_database`.`M`' does not exist.");
+    }
+
+    @Test
+    void testNonExistingDropModel() throws Exception {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
+
+        assertThat(tEnv.dropModel("M", true)).isFalse();
+
+        assertThatThrownBy(() -> tEnv.dropModel("M", false))
+                .isInstanceOf(ValidationException.class)
+                .hasMessage(
+                        "Model with identifier 'default_catalog.default_database.M' does not exist.");
+    }
+
+    @Test
+    void testCreateTemporaryModelFromDescriptor() {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
+        assertTemporaryCreateModelFromDescriptor(tEnv, false);
+    }
+
+    @Test
+    void testCreateTemporaryModelIfNotExistsFromDescriptor() {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
+
+        assertTemporaryCreateModelFromDescriptor(tEnv, true);
+        assertThatNoException()
+                .isThrownBy(() -> tEnv.createTemporaryModel("M", TEST_MODEL_DESCRIPTOR, true));
+
+        assertThatThrownBy(() -> tEnv.createTemporaryModel("M", TEST_MODEL_DESCRIPTOR, false))
+                .isInstanceOf(ValidationException.class)
+                .hasMessage(
+                        "Temporary model '`default_catalog`.`default_database`.`M`' already exists");
+    }
+
+    @Test
+    void testCreateModelWithSameNameIgnoreIfExists() {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
+
+        tEnv.createModel("M", TEST_MODEL_DESCRIPTOR);
+        tEnv.createModel("M", TEST_MODEL_DESCRIPTOR, true);
+    }
+
+    @Test
+    void testListModels() {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
```

Comment:
Am I right that we have this line for every test in that class?
If so why can't we extract it into a separate field and initialize just with `BerforeEach` ?